src/main/java/org/tikv/raw/RawKVClient.java

```diff
@@ -30,16 +34,21 @@
 import org.tikv.kvproto.Kvrpcpb;
 import org.tikv.kvproto.Metapb;
 
-import java.util.*;
-
 public class RawKVClient implements AutoCloseable {
   private static final String DEFAULT_PD_ADDRESS = "127.0.0.1:2379";
   private final TiSession session;
   private final RegionManager regionManager;
+  private final ExecutorCompletionService<Object> completionService;
+  private static final Logger logger = Logger.getLogger(RawKVClient.class);
+
+  private static final int RAW_BATCH_PUT_SIZE = 16 * 1024;
 
   private RawKVClient(String addresses) {
     session = TiSession.create(TiConfiguration.createRawDefault(addresses));
     regionManager = session.getRegionManager();
+    ExecutorService executors =
+        Executors.newFixedThreadPool(session.getConf().getRawClientConcurrency());
+    completionService = new ExecutorCompletionService<>(executors);
```

Comment:
Might better consider using newCachedThreadPool. If user don't use it, no need to pay the resource of it.